docs/blog_post.md

```diff
@@ -0,0 +1,177 @@
+
+The relationships we are looking for are of moderate depth and can take several different forms.
+
+An organization may use different CI/CD systems. We mostly use CircleCI, GitHub Actions and [AWS CodeBuild](https://tech.ovoenergy.com/building-a-secure-ci-cd-pipeline-for-terraform-infrastructure-as-code/), with a dash of Jenkins laying around for good ol' legacy reasons. Most CI/CD systems support user-defined environment variables, but with different twists. For example:
+
+- CircleCI has Contexts (which are away of reusing a bundle of environment variables across project) but also supports a way of directly attaching environment variables to specific Projects (which have a one-to-one mapping with GitHub repositories)
+- GitHub Actions also supports attaching environment variables diretly to repositories, but it also has a notion of Environment (essentially, environment variables directly attached to a repository, but with optional branch protection rules) and organization environment variables (available to all repositories in an organization)
+
+These approaches are quite different. Wouldn't it be great if we could search for all paths between a user and a secret without having to worry about which system(s) we're targetting?
+
+This sounds like a good case for a graph database. It's much more fun to write:
+
+```
+MATCH p=(:User{login:"alice"})-[*..5]->(:EnvironmentVariable)
+RETURN p
+```
+
+than to try to translate all possible paths into an SQL statement.
+
+We opted to work with Neo4j and the Cypher query language; the Community edition is free, easy to use and popular. The folks at Neo4j have also [open-sourced Cypher](https://opencypher.org/) and some [other graph databases](https://docs.aws.amazon.com/neptune/latest/userguide/access-graph-opencypher-bolt.html) have started supporting the Bolt binary protocol.
+
+## What do we ingest?
+
+![all the things](img/ingest-all-the-things.jpg)
+
+Now that we have justified the use of a hipster's DB, what do we store exactly? We'll gloss over this here; you're welcome to check the [code](https://github.com/ovotech/gitoops) and the [schema](https://github.com/ovotech/gitoops/blob/main/docs/schema.md) for the details.
+
+### Access Controls
+
+We want to know:
+
+- Relationships between users and teams within an organization.
+- Repositories that users/teams can access. We're particular interested in repositories a user/team can open a pull request against (`write` access in GitHub) or have admin access on (allowing them to disable branch protections to abuse a production deployment pipeline triggered on a merge to the main branch, for example).
+- Which repositories have branch protections enabled?
+
+### CI/CD Configurations
+
+We want to know which repositories have CI/CD configurations:
+
+- We ingest CI/CD configuration files for popular CI/CD systems.
+- For CircleCI we can query the API to figure out which repositories have a project configured.
+
+### Events
+
+We want to know:
```

Comment:
Are you planning to add more examples here? If not, perhaps remove the top level bullet point?